### PR TITLE
kubectl-graph: init at 0.7.0, add maintainer rksm

### PR DESCRIPTION
**Pull request by rksm**
Add kubectl-graph, a kubectl plugin for visualizing dependencies between kubernetes resources.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x]  Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ku/kubectl-graph/package.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+buildGoModule rec {
+  pname = "kubectl-graph";
+  version = "0.7.0";
+
+  src = fetchFromGitHub {
+    owner = "steveteuber";
+    repo = "kubectl-graph";
+    rev = "v${version}";
+    sha256 = "sha256-5N1eC8J0nHEgFUCHEn5b3kUDj6MWejouQBKkdJKsaAo=";
```
Comment (nevivurn) [inline pkgs/by-name/ku/kubectl-graph/package.nix:15]:
```suggestion
    hash = "sha256-5N1eC8J0nHEgFUCHEn5b3kUDj6MWejouQBKkdJKsaAo=";
```

File: pkgs/by-name/ku/kubectl-graph/package.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  };
+
+  vendorHash = "sha256-fvn+CoOyMRjsIemMRXitMjTlbbhjrlDIHu398b4/ZZ0=";
+
+  meta = with lib; {
+    description = "A kubectl plugin to visualize Kubernetes resources and relationships.";
```
Comment (nevivurn) [inline pkgs/by-name/ku/kubectl-graph/package.nix:21]:
```suggestion
    description = "kubectl plugin to visualize Kubernetes resources and relationships";
```

See https://github.com/NixOS/nixpkgs/tree/master/pkgs#meta-attributes

File: pkgs/by-name/ku/kubectl-graph/package.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  src = fetchFromGitHub {
+    owner = "steveteuber";
+    repo = "kubectl-graph";
+    rev = "v${version}";
+    sha256 = "sha256-5N1eC8J0nHEgFUCHEn5b3kUDj6MWejouQBKkdJKsaAo=";
+  };
+
+  vendorHash = "sha256-fvn+CoOyMRjsIemMRXitMjTlbbhjrlDIHu398b4/ZZ0=";
+
+  meta = with lib; {
```
Comment (nevivurn) [inline pkgs/by-name/ku/kubectl-graph/package.nix:20]: I recommend removing the `with lib;` here and just using `lib.maintainers` etc.

File: maintainers/maintainer-list.nix [hunk 4]
```diff
@@ -18250,6 +18250,12 @@
     githubId = 82817;
     name = "Robert Kreuzer";
   };
+  rksm = {
```
Comment (nevivurn) [inline maintainers/maintainer-list.nix:18253]: The maintainers list change should be in a separate commit, with a message like `maintainers: add rksm`

**Review by nevivurn — CHANGES_REQUESTED** [review]

**Comment by rksm** [comment]
Thank you for the review @nevivurn!

**Review by nevivurn — COMMENTED** [review]
Minor change, the commit message should not include `v` in the version.
Otherwise looks good to me.

**Review by nevivurn — APPROVED** [review]

File: pkgs/by-name/ku/kubectl-graph/package.nix [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    owner = "steveteuber";
+    repo = "kubectl-graph";
+    rev = "v${version}";
+    hash = "sha256-5N1eC8J0nHEgFUCHEn5b3kUDj6MWejouQBKkdJKsaAo=";
+  };
+
+  vendorHash = "sha256-fvn+CoOyMRjsIemMRXitMjTlbbhjrlDIHu398b4/ZZ0=";
+
+  meta = {
+    description = "kubectl plugin to visualize Kubernetes resources and relationships";
```
Comment (FliegendeWurst) [inline pkgs/by-name/ku/kubectl-graph/package.nix:21]:
```suggestion
    description = "Kubectl plugin to visualize Kubernetes resources and relationships";
```

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

**Review by FliegendeWurst — APPROVED** [review]

**Review by nevivurn — COMMENTED** [review]
Please squash the last two commits, should be good to go?

**Review by nevivurn — APPROVED** [review]

**Review by FliegendeWurst — APPROVED** [review]

**Comment by rksm** [comment]
Any chance this can be merged? Happy to add/maintain other packages but my maintainer entry depends on this PR.

**Review by Aleksanaa — APPROVED** [review]
